src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUncommentedMainTest.java

```diff
@@ -0,0 +1,66 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.UncommentedMainCheck;
+
+public class XpathRegressionUncommentedMainTest extends AbstractXpathTestSupport {
+
+    private final Class<UncommentedMainCheck> clazz =
+            UncommentedMainCheck.class;
+
+    @Override
+    protected String getCheckName() {
+        return clazz.getSimpleName();
+    }
+
+    @Test
+    public void testOne() throws Exception {
```

Comment:
maybe this test could have a more descriptive name?

Reply:
we did not find any good names for them, that why it is like this in all Xpath tests.

@esilkensen , do you have suggestions ?

Reply:
Ah, this sounds good to me then - I didn’t realize it was the same in all other tests too.

Reply:
it does not mean it is good in all others :) , we just do not know better alternatives.

reason of this tests is not to validate Xpath correctness and make sure it works in general, that is why names are so general. If in future we will run into problems of xpath for particular Check then we will create test like `testOnInnerClasswithAnnotationAndXxxxxx`.